code/modules/clothing/clothing.dm

```diff
@@ -217,6 +217,23 @@
 		for(var/A in armor)
 			armor[A] -= rand(armor[A]/3, armor[A])
 
+/obj/item/clothing/attack(mob/living/M, mob/living/user, def_zone, var/originator = null)
+	if(iscarbon(user))
+		if(user.a_intent == I_HELP)
+			if(istype(src, /obj/item/clothing/under) || (istype(src, /obj/item/clothing/suit) && !istype(src, /obj/item/clothing/suit/space)))
+				if(ishuman(M) && M.on_fire)
+					var/mob/living/carbon/human/target = M
+					if(isplasmaman(target)) // Cannot put out plasmamen, else they could just go around with a jumpsuit and not need a space suit.
+						visible_message("<span class='warning'>\The [user] attempts to put out the fire on \the [target], but plasmafires are too hot. It is no use.</span>")
+						return
```

Comment:
While it's fine to return both here and under the `else `branch, you could just return after the `if` only once.

Reply:
Without the `return` under the `isplasmaman` branch we would call ..() and hit the plasmaman with our piece of clothing, I think.

Reply:
Correct, I did a dumb.